scripts/imgtool/design.md

```diff
@@ -244,6 +245,30 @@ work properly even when it is reset during the middle of an image swap. For this
 reason, the rest of the document describes its behavior when configured to swap
 images during an upgrade.
 
+### [RAM Loading](#ram-loading)
+
+In ram-loading mode the slots are equal. Like the direct-xip mode, this mode
+also selects the newest image by reading the image version numbers in the image
+headers. But instead of executing it in place, the newest image is copied to the
+RAM for execution. The load address, the location in RAM where the image is
+copied to, is stored in the image header. The ram-loading upgrade mode can be
+useful when there is no internal flash in the SoC, but there is a big enough
+internal RAM to hold the images. Usually in this case the images are stored
+in an external storage device. Execution from external storage has some
+drawbacks (lower execution speed, image is exposed to attacks) therefore the
+image is always copied to the internal RAM before the authentication and
+execution. Ram-loading mode requires the image must be built to be executed
+from the RAM address range instead of the storage device address range. If
+ram-loading is enabled then platform must define the following parameters:
+
+```c
+#define IMAGE_EXECUTABLE_RAM_START    <area_base_addr>
+#define IMAGE_EXECUTABLE_RAM_SIZE     <area_size_in_bytes>
+```
+
+The ram-loading mode currently supports only the single image boot and the image
```

Comment:
The single image boot sound ambiguous;y to me (either only one image or dual bank mode).
It is general comment `Multi-mage mode` should be rephrased to something like `Multi-image simultaneous execution mode `.

Reply:
I suggest to handle this renaming in a separate change, since it is independent from this feature.

Reply:
We have [SINGLE_IMAGE_DFU](https://github.com/JuulLabs-OSS/mcuboot/blob/master/boot/zephyr/Kconfig#L61) option for zephyr - maybe `dual bank boot`?

Reply:
I've created a new issue to discuss this topic: https://github.com/JuulLabs-OSS/mcuboot/issues/800.
I would rather rename the `MCUBOOT_SINGLE_IMAGE_DFU` option to something like `MCUBOOT_SINGLE_SLOT_DFU`.

Reply:
Good idea!